File: medium/2181-Merge_Nodes_in_Between_Zeros/Merge_nodes_in_between_zeros.cpp
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     ListNode *next;
 *     ListNode() : val(0), next(nullptr) {}
 *     ListNode(int x) : val(x), next(nullptr) {}
 *     ListNode(int x, ListNode *next) : val(x), next(next) {}
 * };
 */

class Solution {
public:
    ListNode* mergeNodes(ListNode* head) {
               
        ListNode *output = new ListNode();
        ListNode *pointer = output; 
        
        int number = 0;
       
        while (head = head->next) {
            
            if (!head->next){
                pointer->val = number;
                return output;
            }

            if (!head->val) {
                pointer->val = number;
                // Get new node
                pointer->next = new ListNode();
                pointer = pointer->next;
                number = 0;
            }else {
                number += head->val;
            }
        }
        return output;
    }
};

File: medium/2181-Merge_Nodes_in_Between_Zeros/Merge_nodes_in_between_zeros.py
from typing import Optional
from utils.linked_list import *
class Solution:
    def mergeNodes(self, head: Optional[ListNode]) -> Optional[ListNode]:

        number = 0
        output = ListNode()
        pointer = output

        while head.next:
            head = head.next
            if head.val == 0:
                if head.next:
                    pointer.val = number
                    pointer.next = ListNode()
                    pointer = pointer.next
                    number = 0
                else:
                    pointer.val = number
            else:
                number += head.val

        return output